sql/hive-thriftserver/src/main/scala/org/apache/spark/sql/hive/thriftserver/SparkSQLCLIDriver.scala

```diff
@@ -519,15 +519,29 @@ private[hive] class SparkSQLCLIDriver extends CliDriver with Logging {
   // Note: [SPARK-31595] if there is a `'` in a double quoted string, or a `"` in a single quoted
   // string, the origin implementation from Hive will not drop the trailing semicolon as expected,
   // hence we refined this function a little bit.
+  // Note: [SPARK-33100] Ignore the content inside bracketed comment and ignore the comment without
+  // content.
   private def splitSemiColon(line: String): JList[String] = {
     var insideSingleQuote = false
     var insideDoubleQuote = false
-    var insideComment = false
+    var insideDashComment = false
+    var insideBracketedComment = false
+    var bracketedCommentRightBound = -1
     var escape = false
     var beginIndex = 0
+    var contentBegin = false
     val ret = new JArrayList[String]
 
+    def insideComment: Boolean = insideDashComment || insideBracketedComment
+    def isContent(char: Char): Boolean = !insideComment && !s"$char".trim.isEmpty
```

Comment:
`!s"$char".trim.isEmpty` -> `char != ' '`?

Reply:
this character might be a "\t" or "\n".